Coverage improvements (#2477)
foundry-rs/starknet-foundry: docs/src/testing/coverage.md

## Prerequisites

> cairo-coverage relies on debug information provided by Scarb. To generate needed debug information, you need to have:

Reviewer: nit:
```suggestion
`cairo-coverage` relies on debug information provided by Scarb. To generate the necessary debug information, you need to have:
```

> `cairo-coverage` relies on debug information provided by Scarb. To generate the necessary debug information, you need to have:

Reviewer 2: ```suggestion
`cairo-coverage` relies on debug information provided by Scarb. To generate the necessary debug information, you need to install:
```

Reviewer: `install` might not be the most accurate in this context given that the 2. point is about configuring `Scarb.toml`

```shell
$ snforge test --coverage
```

This will generate a coverage report in the `coverage` directory named `coverage.lcov`.

> ## Coverage report

Reviewer: I am not sure that this section is crucial in the forge documentation. However, if we want to include it here I suggest to modify wording a bit. As it is currently written, it seems to imply generating a coverage report from another coverage report, which may be confusing for readers.

> You can generate a coverage report from many tools that accept the `lcov` format.

Reviewer 2: ```suggestion
`cairo-coverage` only generates `.lcov` files. Coverage report can be generated by using one of many tools that accept the `lcov` format.
```

In this example we will use the `genhtml` tool from the `lcov` package to generate an HTML report.

> On your coverage file run:

Reviewer 2: ```suggestion

Run the following command in the directory containing your `coverage.lcov` file:
```

```shell

> genhtml -o coverage_report coverage.lcov

Reviewer 2: ```suggestion
$ genhtml -o coverage_report coverage.lcov
```

```

> You can now open the `index.html` file in the `coverage_report` directory to see the coverage report.

Reviewer 2: ```suggestion
You can now open the `index.html` file in the `coverage_report` directory to see the generated coverage report.
```

```shell
$ snforge test --coverage
```

This will generate a coverage report in the `coverage` directory named `coverage.lcov`.

## Coverage report

> `cairo-coverage` only generates `.lcov` files. Coverage report can be generated by using one of many tools that accept the `lcov` format.

Reviewer: ```suggestion
`cairo-coverage` generates coverage data as an `.lcov` file. A summary report with aggregated data can be produced by one of many tools that accept the `lcov` format.
```

> In this example we will use the `genhtml` tool from the `lcov` package to generate an HTML report.

Reviewer 2: Let's add some relevant link to this package
